### PR TITLE
Add integration test framework

### PR DESCRIPTION
**Pull request by ekoby**
Uses:
- CMake/CTest
- ziti CLI quickstart
- expect

### CODE REVIEW

File: tests/integ/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+)
+
+if (WIN32)
+    set_property(TARGET integ-tests PROPERTY CXX_STANDARD 20)
+else ()
+    set_property(TARGET integ-tests PROPERTY CXX_STANDARD 14)
+endif ()
+
+add_custom_target(ziti-cli ALL
+        COMMAND ${CMAKE_COMMAND} -E env GOBIN=${CMAKE_CURRENT_BINARY_DIR} ${GOLANG_EXE} install -v github.com/openziti/ziti/ziti@v0.31
```
[the remark this answers is not in the document]
Comment (dovholuknf) [reply tests/integ/CMakeLists.txt:24]: actually, we'd know the arch too now-a-days and could just fetch the proper one . I'd rather see us use 
```
    FetchContent_Declare(
            ziti-exe
            URL "https://github.com/openziti/ziti/releases/download/....blahblahblah"
    )
    FetchContent_MakeAvailable(ziti-exe)
```

[the remark this answers is not in the document]
Comment (ekoby) [reply tests/integ/CMakeLists.txt:24]: do you really think that if a developer is working on ZitiSDK having install golang is a burden?>?

[the remark this answers is not in the document]
Comment (michaelquigley) [reply tests/integ/CMakeLists.txt:24]: for anyone doing C development, having to install golang is probably the least annoying thing they have to deal with. :-)

[the remark this answers is not in the document]
Comment (ekoby) [reply tests/integ/CMakeLists.txt:24]: ... and it does NOT apply if you're just consuming Ziti SDK

**Pull request merged**